dali/pipeline/operator/builtin/external_source.h

```diff
@@ -287,6 +297,22 @@ class ExternalSource : public Operator<Backend>, virtual public BatchSizeProvide
   DISABLE_COPY_MOVE_ASSIGN(ExternalSource);
 
  protected:
+  inline bool HasNdim() {
+    return !layout_.empty() || spec_.HasArgument("ndim");
+  }
+
+  inline void InferNdim() {
+    if (!layout_.empty()) {
```

Comment:
:+1: Keep it like that even if we get rid of the default - this way we can still use empty layouts, which has some merit at our Python interop level (namely, we tend to return an empty string rather than `None`).

Reply:
I guess there's no other way than treat empty as no layout.